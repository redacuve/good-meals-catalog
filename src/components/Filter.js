import React from 'react';
import PropTypes from 'prop-types';

function Filter(props) {
  const { categories, changeFilter } = props;
  return (
    <select onChange={e => changeFilter(e.target.value)}>
      <option value="All">All</option>
      {categories.map(c => (
        <option key={c} value={c}>
          {c}
        </option>
      ))}
    </select>
  );
}

Filter.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  changeFilter: PropTypes.func.isRequired,
};

export default Filter;
