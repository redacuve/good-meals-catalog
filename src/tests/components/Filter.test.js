import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Filter from '../../components/Filter';

afterEach(cleanup);
const categories = ['Beef', 'Breakfast', 'Chicken', 'Dessert', 'Goat', 'Lamb', 'Miscellaneous', 'Pasta', 'Pork', 'Seafood', 'Side', 'Starter', 'Vegan', 'Vegetarian'];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Filter categories={categories} changeFilter={() => {}} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render Filter correctly', () => {
  const { getByTestId } = render(<Filter categories={categories} changeFilter={() => {}} />);
  expect(getByTestId('filter-list')).toHaveTextContent(/^AllBeefBreakfastChickenDessertGoatLambMiscellaneousPastaPorkSeafoodSideStarterVeganVegetarian$/);
});

it('render Filter Message correctly', () => {
  const { getByTestId } = render(<Filter categories={categories} changeFilter={() => {}} />);
  expect(getByTestId('filter-list')).toContainHTML('<select class="appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline" data-testid="filter-list"><option value="All">All</option><option value="Beef">Beef</option><option value="Breakfast">Breakfast</option><option value="Chicken">Chicken</option><option value="Dessert">Dessert</option><option value="Goat">Goat</option><option value="Lamb">Lamb</option><option value="Miscellaneous">Miscellaneous</option><option value="Pasta">Pasta</option><option value="Pork">Pork</option><option value="Seafood">Seafood</option><option value="Side">Side</option><option value="Starter">Starter</option><option value="Vegan">Vegan</option><option value="Vegetarian">Vegetarian</option></select>');
});

it('matches snapshot 1', () => {
  const tree = renderer.create(<Filter categories={categories} changeFilter={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
