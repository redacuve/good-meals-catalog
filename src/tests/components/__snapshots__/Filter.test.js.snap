// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="inline-block relative w-full"
>
  <select
    className="appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
    data-testid="filter-list"
    onChange={[Function]}
  >
    <option
      value="All"
    >
      All
    </option>
    <option
      value="Beef"
    >
      Beef
    </option>
    <option
      value="Breakfast"
    >
      Breakfast
    </option>
    <option
      value="Chicken"
    >
      Chicken
    </option>
    <option
      value="Dessert"
    >
      Dessert
    </option>
    <option
      value="Goat"
    >
      Goat
    </option>
    <option
      value="Lamb"
    >
      Lamb
    </option>
    <option
      value="Miscellaneous"
    >
      Miscellaneous
    </option>
    <option
      value="Pasta"
    >
      Pasta
    </option>
    <option
      value="Pork"
    >
      Pork
    </option>
    <option
      value="Seafood"
    >
      Seafood
    </option>
    <option
      value="Side"
    >
      Side
    </option>
    <option
      value="Starter"
    >
      Starter
    </option>
    <option
      value="Vegan"
    >
      Vegan
    </option>
    <option
      value="Vegetarian"
    >
      Vegetarian
    </option>
  </select>
  <div
    className="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
  >
    <svg
      className="fill-current h-4 w-4"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
      />
    </svg>
  </div>
</div>
`;
